### DISCUSSION TITLE
when is actionscript 3 supposed to be released?

### DISCUSSION THREAD

**Discussion by narwalions** [ruffle-rs/ruffle#5673] [Q&A] [answered]
I know that ruffle is very work in progress, but even a very rough estimate would be greatly appreciated. yes, I know it may be as accurate as getting a caves and cliffs update. if all else fails, someone just gives me a random date so I have something to look forewards to.

**Comment by Pythone** [comment]
I am not sure what to tell you. However, the best swf-to-js converter to do AS3 is Mozilla Shumway. I think it would be faster if Ruffle takes inspiration from Shumway. LightSpark is a Free and Open Source flash player that is the best AS3 player. We do not have a date.

**Comment by Poopooracoocoo** [comment, marked as answer]
1. Ruffle is not Minecraft. Mojang has paid developers who work full time on it, Ruffle does not. They can therefore give you an ETA.
2. As such, developers of Ruffle do so in their free time.
3. It's going to take a long time. Try 1/12/2022? :P
